shared/source/os_interface/linux/drm_memory_manager.cpp

```diff
@@ -341,29 +339,10 @@ bool DrmMemoryManager::setSharedSystemAtomicAccess(const void *ptr, const size_t
     auto &drm = this->getDrm(rootDeviceIndex);
     auto ioctlHelper = drm.getIoctlHelper();
 
-    uint32_t attribute = 0;
-    uint64_t param = 0;
+    uint32_t attribute = ioctlHelper->getAtomicAdvise(false);
+    uint32_t atomicParam = ioctlHelper->getAtomicAccess(mode);
 
-    switch (mode) {
-    case AtomicAccessMode::device:
-        attribute = ioctlHelper->getAtomicAdvise(false);
-        param = (static_cast<uint64_t>(ioctlHelper->getDrmParamValue(DrmParam::atomicClassDevice)) << 32);
-        break;
-    case AtomicAccessMode::system:
-        attribute = ioctlHelper->getAtomicAdvise(false);
-        param = (static_cast<uint64_t>(ioctlHelper->getDrmParamValue(DrmParam::atomicClassGlobal)) << 32);
-        break;
-    case AtomicAccessMode::host:
-        attribute = ioctlHelper->getAtomicAdvise(false);
-        param = (static_cast<uint64_t>(ioctlHelper->getDrmParamValue(DrmParam::atomicClassSystem)) << 32);
-        break;
-    case AtomicAccessMode::none:
-        attribute = ioctlHelper->getAtomicAdvise(false);
-        param = (static_cast<uint64_t>(ioctlHelper->getDrmParamValue(DrmParam::atomicClassUndefined)) << 32);
-        break;
-    default:
-        return false;
-    }
+    uint64_t param = (static_cast<uint64_t>(atomicParam) << 32) | 0;
```

Comment:
why we need to do those shifts after returning the value from iocthHelper?
IoctlHelper should rather do this translation internally if needed.
Also what is the point of | 0 ?

Reply:
you can change it to 64 bit

Reply:
we need it anyway, as right now you need to do those shifts, which requires helper function